tutorials/add_model.md

```diff
@@ -0,0 +1,139 @@
+
+    def fit(self, train_set):
+        """Fit the model to observations.
+
+        Parameters
+        ----------
+        train_set: object of type TrainSet, required
+            An object contraining the user-item preference in csr scipy sparse format,\
+            as well as some useful attributes such as mappings to the original user/item ids.\
+            Please refer to the class TrainSet in the "data" module for details.
+        """
+
+    def score(self, user_id, item_id=None):
+        """Predict the scores/ratings of a user for an item.
+
+        Parameters
+        ----------
+        user_id: int, required
+            The index of the user for whom to perform score prediction.
+
+        item_id: int, optional, default: None
+            The index of the item for that to perform score prediction.
+            If None, scores for all known items will be returned.
+
+        Returns
+        -------
+        res : A scalar or a Numpy array
+            Relative scores that the user gives to the item or to all known items
+        """
+``` 
+Every model is an extension of the `Recommender` class. All you need to do is to redefine the functions listed in the above `recom_pmf.py` file.
+
+The `fit()` function should contain or make a call to the necessary codes to fit your model to the training data, while the `score()` function specifies how to perform predictions using your model.
+ 
+#### More on the score function 
+When you perform rating prediction evaluation (e.g., in terms of RMSE) Cornac calls the `Recommender.rate()` function, which in turn calls the `score()` function and clips the predicted values to lie in the same range as the original ratings. Analogously, Cornac calls the `Recommender.rank()` function, which returns a ranking of items according to their scores as predicted by the function `score()`. We recommend you to take a look at the `rate()` and `rank()` functions inside the `Recommender` class, as you may wish to redefine them as well, if the default settings do not meet your needs. 
+
+### 2. Making your recommender available to Cornac
+As you may have already noticed, Cornac treats each recommender model as a separate module, so as to reduce coupling and ease the contribution of new models. This is the reason why you need to include a `pmf/__init__.py` file:
+```python
+from .recom_pmf import PMF
+```
+
+One last step is required to make you recommender model available to Cornac, which is to update `../models/__init__.py` by adding the following line:
+```python
+from .pmf import PMF
+```
+
+### 3. Indicating dependencies
+The file `requirements.txt` is optional as opposed to the others, and it is only needed if your implementation relies on some external libraries/frameworks, e.g., TensorFlow, PyTorch, etc., in which case you should include this file and indicate which versions of the framework(s) are required. Here is a sample of a `requirements.txt` file:
```

Comment:
The file `requirements.txt` is optional as opposed to the others, and it is only needed if your implementation relies on some supporting libraries (e.g., TensorFlow, PyTorch) which are not Cornac's dependencies. In that case, you should include this file and indicate which versions of the libraries are required as for future users to refer to. Here is a sample of a `requirements.txt` file:

Reply:
Thanks, I will rephrase this part a little bit.